packages/java/src/node/java-util.ts

```diff
@@ -0,0 +1,81 @@
+/********************************************************************************
+ * Copyright (C) 2018 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { NodePackage, ExtensionPackage } from '@theia/application-package';
+import { ApplicationPackageManager } from '@theia/application-manager';
+import { injectable } from 'inversify';
+import * as paths from 'path';
+
+/**
+ * Represents contributes section from package.json.
+ * Contributes section describes paths to the java bundles and looks like:
+ * "contributes": {
+ *   "javaExtensions": ["./server/my.java.plugin.jar"],
+ * }
+ */
+interface JavaContributes extends NodePackage {
+    /**
+     * An array of paths to the bundles.
+     */
+    javaExtensions?: string[] | undefined;
+}
+
+/**
+ * The collector to find all java bundles in Theia extensions.
+ */
+@injectable()
+export class ExtensionBundlesCollector {
+    private javaExtensions: string[] | undefined;
+
+    /**
+     * Collects paths of all java bundles declared in the extensions.
+     *
+     * @return an array of paths to the bundles
+     */
+    public collectExtensionBundlesPaths(): string[] {
+        if (this.javaExtensions) {
+            return this.javaExtensions;
+        }
+
+        const projectPath = process.cwd();
```

Comment:
As I understand this code, in the Java bundle, I would have to declare other bundles who can extend jdt.ls? This is not what we want, we want to have unknown third party bundles contribute bundles to jdt.ls. We should not be reading other extensions package.json by hand, we should declare an extension point, which other extensions can contribute to and then have the system pass us those extensions.

Reply:
@tsmaeder it's not that. You don't need to declare bundles in Java extension if you want to extend jdt.ls. 
Java extension in Theia just contains a mechanism for scanning all other Theia extensions and if some of them has java bundle (plugin for jdt.ls) it takes a path to this bundle (which was described in packaje.json in this extension) and adds this path to `initializationOptions`. And when jdt.ls is starting all found bundles from Theia extensions will be installed as well.

Imagine this:
-  Theia has **A** extension. 
-  This extension contains bundle (**custom-java-plugin.jar**) to jdt.ls (in some subfolder)
-  `package.json` in **A** contains the path to the bundle

So, what Java extension does:
-  Before initializing jdt.ls Java extension scans all other Theia extensions
-  It finds that extension **A** has declaration about jdt.ls plugin
-  Builds absolute path to this plugin (/..../A/subfolder/custom-java-plugin.jar)
-  Adds this path to `initializationOptions` and initializes  jdt.ls
-  jdt.ls loads **custom-java-plugin.jar** bundle and installs it.